### PR TITLE
Api2 data sources

### PR DESCRIPTION
**Pull request by mknblch**
Adds a datasource for a single relationship type which implements iteration in one direction.

### CODE REVIEW

File: core/src/main/java/org/neo4j/graphalgo/core/utils/RelationshipContainer.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+         * set the builder to aim for nodeId and initialize its underlying connection array
+         *
+         * @param nodeId the current (mapped) nodeId
+         * @param degree the degree for the node
+         * @return this instance for method chaining
+         */
+        public Builder aim(int nodeId, int degree) {
+            this.nodeId = nodeId;
+            offset = 0;
+            container.data[nodeId] = new int[degree];
```
Comment (jexp) [inline core/src/main/java/org/neo4j/graphalgo/core/utils/RelationshipContainer.java]: could the degree also be inferred upon adding new targetNodeIds ? (at least as an option?

Comment (mknblch) [inline core/src/main/java/org/neo4j/graphalgo/core/utils/RelationshipContainer.java]: I could add a logic which grows the array on demand.

Comment (jexp) [inline core/src/main/java/org/neo4j/graphalgo/core/utils/RelationshipContainer.java]: Oh sorry, I thought this was just keeping track of the degree counter.
If we have the data upfront, it's fine to pre-size it.

Comment (mknblch) [inline core/src/main/java/org/neo4j/graphalgo/core/utils/RelationshipContainer.java]: Ok - But you are right. We might need an autogrowing implementation in future for extracting paths (e.g. shortest path forest)

Comment (mknblch) [inline core/src/main/java/org/neo4j/graphalgo/core/utils/RelationshipContainer.java]: Or perhaps we find an even better struct for this case. I made issue #69 out of it but im going to delay it until we need it.

**Pull request merged**